internal/core/src/exec/expression/NullExpr.cpp

```diff
@@ -88,6 +88,41 @@ PhyNullExpr::ExecVisitorImpl(OffsetVector* input) {
     if (auto res = PreCheckNullable(input)) {
         return res;
     }
+    // when T is ArrayView, the ScalarIndex<T> shall be ScalarIndex<ElementType>
+    // NOT ScalarIndex<ArrayView>
+    if (is_index_mode_ && std::is_same_v<T, ArrayView>) {
```

Comment:
This change pull the is index judgement out of `ProcessChunksForValid`, but still leave the risk that other expr might experience this.
Let's change `ProcessChunksForValid`'s signature to accept view type to force all expr will follow this pattern.

BTW this element type need to be a template of view, bad code structure, anyway